### ISSUE TITLE
Remove chance_of_loss (sidx -4): computed, carried through fmpy, then discarded

### ISSUE THREAD

**Issue by sstruzik** [OasisLMF/OasisLMF#2143]
## Summary

`chance_of_loss` (sidx `-4`) is computed by the ground-up tools, carried through the financial
module as a pass-through, and then **discarded**. Nothing in the pytools pipeline consumes it.
It should be removed from the computation and from the stream.

## Evidence that nothing reads it

Traced end to end:

| stage | what happens to `-4` |
|---|---|
| `gulpy` / `gulmc` | computed in `compute_mean_loss` (`gul/core.py:181`) and written |
| `plapy` | iterated over; deliberately **not** scaled since #<this PR> |
| `fmpy` | stored in `pass_through`, re-emitted unchanged (`fm/stream_sparse.py:382`) |
| `summarypy` | **dropped** — `summary/manager.py:219` accumulates only `sidx > 0` and `-1`, `-3`, `-5` |
| `aalpy` / `lecpy` / `pltpy` | read `-4` as the **affected-risk count**, which `summarypy` writes into the same slot |
| `eltpy` | does not reference it |

The `ChanceOfLoss` column in MELT/MPLT output does **not** come from this value — `eltpy` and
`pltpy` compute it themselves from the sample losses:

```python
chance_of_loss = state["non_zero_samples"] / state["len_sample"]   # elt/manager.py:242
state["chance_of_loss"] += (loss > 0) / state["len_sample"]        # plt/manager.py:416
```

So the special is dead data occupying a record per item, per event, in every ground-up and
financial-module stream.

## What removal would involve

- `gul/core.py` — drop `chance_of_loss` from `compute_mean_loss` and its return tuple
- `gul/manager.py`, `gulmc/manager.py` — drop the unpack and the `losses[CHANCE_OF_LOSS_IDX]` write
- `gul/common.py` — drop `-4` from `SPECIAL_SIDX`. **Careful**: `NUM_IDX = SPECIAL_SIDX.shape[0]`
  sizes the `losses` array, which is addressed by the negative constants, so `NUM_IDX` must stay
  at 5 or `MAX_LOSS_IDX == -5` starts pointing at the wrong row.
- `fm/` — remove the `pass_through` machinery entirely (~28 references across `manager.py`,
  `stream_sparse.py`, `compute_sparse.py`), which exists solely to carry this one value. This is
  the bulk of the win.

## Why this is an issue and not a PR

It is a **stream format change**, and **1,034 of 1,218** committed `.bin` reference fixtures
contain a `-4` record (`test_model_1..7`, 144 each, plus `test_summarypy`). Regenerating them all
means they are produced by the very code being changed, so they stop independently verifying
anything about the removal — which is precisely the safety net this kind of edit needs.

Suggested sequencing:

1. Make the change.
2. Diff old vs new streams **record by record** and prove the only difference is the absence of
   `-4` records — do not rely on regenerated fixtures for this.
3. Only then regenerate, with the diff evidence in the PR.

## Open question before starting

Does anything **outside** this repo consume `-4` from a raw ground-up or financial-module stream?
The Oasis Platform, a customer's own consumer, or tooling reading a `bintocsv` dump. Inside
pytools the answer is no, but this is a public stream format, so it is worth confirming before
removing it.

## Context

Found while reviewing the building-packing branch. Related fix already made: `plapy` was
multiplying `-4` by the amplification factor, turning a probability of 0.8 into 1.6. That is
correct now regardless of whether this issue is actioned.

Under building packing the waste is larger: a packed item emits one chance-of-loss **per
building**, which the financial module collapses back to one and then everything downstream
discards.
